### ISSUE TITLE
Mangling breaks code where multiple binding for a var name in function

### ISSUE THREAD

**Issue by overlookmotel** [overlookmotel/livepack#494]
`mangle` option causes var names to be shortened. However, it avoids altering var names where they're the name of a function, to avoid changing the function's `.name` property.

e.g.:

Input:

```js
export default function outer() {
  let someVar;
  function fn() {}
  return [someVar, fn];
});
```

Serialized output:

```js
export default function outer() {
  let a;
  function fn() {}
  return [a, fn];
};
```

This can malfunction when a var name is used for two different bindings in the outer function:

```js
export default function outer() {
  function fn() {}
  {
     let fn;
  }
  return () => fn;
}
```

Serialized output:

```js
export default function outer() {
  function fn() {}
  {
    let a;
  }
  return () => a;
}
```

The problem occurs when external vars of the arrow function are merged into internal vars of outer function.

https://github.com/overlookmotel/livepack/blob/fc4fa3b99e769a23fbf2df36cb68e60e882bebe8/lib/serialize/parseFunction.js#L878-L882

This relies on `internalVars` being undefined for vars which are function names. But in this case, the `let fn` binding means `internalVars.fn` is defined.

2 possible solutions:

1. External vars in `scope.vars` have an `isFunction` property. If it's set, those external vars are not converted to internal vars.
2. Internal vars record the block IDs of the internal bindings (excluding bindings which are function names). If block ID of external var is not in that list of block IDs, it's not converted to an internal var.

**Comment by overlookmotel** [comment]
Solutions using the 2 options above on [issue-494](https://github.com/overlookmotel/livepack/tree/issue-494) and [issue-494-v2](https://github.com/overlookmotel/livepack/tree/issue-494-v2) branches.

[issue-494-v2](https://github.com/overlookmotel/livepack/tree/issue-494-v2) is simpler, but I'm not completely sure it'll always work. If external vars are created in 1st pass, they could end up with incorrect `isFunction` if defined as a `var` first, and then by a function declaration. Need to check this.

**Comment by overlookmotel** [comment]
Commit which originally closed this 3806394 was later modified. edf1b6d is the commit which actually closed this.
